dataline-server/src/main/java/io/dataline/server/apis/ConfigurationApi.java

```diff
@@ -116,177 +121,194 @@ public ConfigurationApi(final Path dbRoot, BasicDataSource connectionPool) {
-    return schedulerHandler.checkDestinationImplementationConnection(
-        destinationImplementationIdRequestBody);
+    return execute(() -> schedulerHandler.checkDestinationImplementationConnection(destinationImplementationIdRequestBody));
   }
 
   // CONNECTION
 
   @Override
   public ConnectionRead createConnection(@Valid ConnectionCreate connectionCreate) {
-    return connectionsHandler.createConnection(connectionCreate);
+    return execute(() -> connectionsHandler.createConnection(connectionCreate));
   }
 
   @Override
   public ConnectionRead updateConnection(@Valid ConnectionUpdate connectionUpdate) {
-    return connectionsHandler.updateConnection(connectionUpdate);
+    return execute(() -> connectionsHandler.updateConnection(connectionUpdate));
   }
 
   @Override
   public ConnectionReadList listConnectionsForWorkspace(@Valid WorkspaceIdRequestBody workspaceIdRequestBody) {
-    return connectionsHandler.listConnectionsForWorkspace(workspaceIdRequestBody);
+    return execute(() -> connectionsHandler.listConnectionsForWorkspace(workspaceIdRequestBody));
   }
 
   @Override
   public ConnectionRead getConnection(@Valid ConnectionIdRequestBody connectionIdRequestBody) {
-    return connectionsHandler.getConnection(connectionIdRequestBody);
+    return execute(() -> connectionsHandler.getConnection(connectionIdRequestBody));
   }
 
   @Override
   public ConnectionSyncRead syncConnection(@Valid ConnectionIdRequestBody connectionIdRequestBody) {
-    return schedulerHandler.syncConnection(connectionIdRequestBody);
+    return execute(() -> schedulerHandler.syncConnection(connectionIdRequestBody));
   }
 
   // JOB HISTORY
 
   @Override
   public JobReadList listJobsFor(@Valid JobListRequestBody jobListRequestBody) {
-    return jobHistoryHandler.listJobsFor(jobListRequestBody);
+    return execute(() -> jobHistoryHandler.listJobsFor(jobListRequestBody));
   }
 
   @Override
   public JobInfoRead getJobInfo(@Valid JobIdRequestBody jobIdRequestBody) {
-    return jobHistoryHandler.getJobInfo(jobIdRequestBody);
+    return execute(() -> jobHistoryHandler.getJobInfo(jobIdRequestBody));
   }
 
   // WEB BACKEND
 
   @Override
   public WbConnectionReadList webBackendListConnectionsForWorkspace(@Valid WorkspaceIdRequestBody workspaceIdRequestBody) {
-    return webBackendConnectionsHandler.webBackendListConnectionsForWorkspace(workspaceIdRequestBody);
+    return execute(() -> webBackendConnectionsHandler.webBackendListConnectionsForWorkspace(workspaceIdRequestBody));
   }
 
   @Override
   public WbConnectionRead webBackendGetConnection(@Valid ConnectionIdRequestBody connectionIdRequestBody) {
-    return webBackendConnectionsHandler.webBackendGetConnection(connectionIdRequestBody);
+    return execute(() -> webBackendConnectionsHandler.webBackendGetConnection(connectionIdRequestBody));
+  }
+
+  private <T> T execute(HandlerCall<T> call) {
```

Comment:
this seems pretty crufty to me. is this a common practice with jersey? i haven't found much in the way of custom error handling that is satisfying.

it would be nice if we could just get the generated code to have reasonable exceptions in the method signature. i looked a little quickly and couldn't figure out how to do it but it is something i plan to look into more.

Reply:
I am not the biggest fan of this design either. the thing I like about it though is that it centralizes were exceptions are transformed into responses.